### ISSUE TITLE
Optimize morpheme training performance (16min/epoch is too slow)

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#53]
## Problem

Training is extremely slow for the morpheme-aware model:
- **16-18 minutes per epoch** for only 10,000 samples
- This is impractical for larger datasets or longer training runs

## Training Details

From `models/morpheme_aware/training.log`:
```
Epoch 1/3 - Loss: 6.9607  (16 minutes)
Epoch 2/3 - Loss: 7.2446  (16 minutes)
Epoch 3/3 - Loss: 7.2832  (18 minutes)
```

## Possible Causes

1. **Inefficient data loading** - no DataLoader optimization
2. **Inefficient batching** - batch size 16 is very small
3. **CPU-only training** - no GPU acceleration
4. **Inefficient masked sampling** - 15% masking computed per batch
5. **Vocabulary lookups** - string-based lookups in hot path

## Investigation Steps

1. Profile training loop to identify bottlenecks
2. Check DataLoader settings (num_workers, pin_memory, prefetch_factor)
3. Measure batch processing time vs data loading time
4. Test larger batch sizes (32, 64, 128)
5. Optimize vocabulary lookups (use integer IDs throughout)

## Success Criteria

- [ ] Training time reduced to <5 minutes per epoch (3x speedup)
- [ ] Identify and document main bottleneck
- [ ] Implement at least 2 optimization improvements

## Priority

P2 - Important but not blocking (can train overnight, but wastes time)

## References

- Training script: `scripts/train_morpheme_aware.py`
- Related: Issue #52 (debug loss), Issue #51 (weight tying)

**Comment by marctjones** [comment]
Superseded by Issue #57 - performance issues irrelevant with wrong objective
